WSLGd/main.cpp

```diff
@@ -186,6 +191,37 @@ try {
 
     SetupOptionalEnv();
 
+    // "ulimits -c unlimited" for core dumps.
+    {
+        struct rlimit limit;
+        limit.rlim_cur = RLIM_INFINITY;
+        limit.rlim_max = RLIM_INFINITY;
+        THROW_LAST_ERROR_IF(setrlimit(RLIMIT_CORE, &limit) < 0);
+    }
+
+    // create folder to store core files.
+    {
+        std::filesystem::create_directories(c_coreDir);
+        THROW_LAST_ERROR_IF(chmod(c_coreDir, 0777) < 0);
+    }
+
+    // update core_pattern.
+    {
+        wil::unique_file file_corePatternFile(fopen(c_corePatternFile, "w"));
+        THROW_LAST_ERROR_IF(!file_corePatternFile.get());
```

Comment:
Maybe don't make this error fatal just in case?

Reply:
ok, I can change that.